haskell/private/unix/unix_nixpkgs.bzl

```diff
@@ -0,0 +1,116 @@
+load("@bazel_tools//tools/cpp:lib_cc_configure.bzl", "get_cpu_value")
+load(
+    "@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl",
+    "nixpkgs_package",
+)
+
+def _unix_nixpkgs(name, packages, **kwargs):
+    nixpkgs_package(
+        name = name,
+        nix_file_content = """
+with import <nixpkgs> {{ config = {{}}; overlays = []; }};
```

Comment:
`rules_nixpkgs` does not inherit `NIX_PATH` instead it constructs its own `NIX_PATH` based on the `repositories` or `repository` attribute. I based this code on the `nixpkgs_cc_configure` rule which also imports `<nixpkgs>`. That one does indeed `buildEnv`, so that might be a better approach.

Reply:
I didn't know that `nixpkgs_cc_configure` was using `<nixpkgs>`. Let's keep it that way then